src/LightSpeed/userFunction.js

```diff
@@ -128,10 +133,18 @@ var LSUserFunction = GObject.registerClass({
 
     bindGlobal(model) {
         this._global = model;
-        this._globalNotifyHandler = model.connect('notify', this._onGlobalNotify.bind(this));
+        const {perLevel, modelProp} = USER_FUNCTIONS[this._codeview.userFunction];
 
-        this._currentLevel = model.currentLevel;
-        this._bindModel(this._global.getModel(this._currentLevel));
+        if (perLevel) {
+            this._globalNotifyHandler = model.connect('notify',
+                this._onGlobalNotify.bind(this));
+            this._currentLevel = model.currentLevel;
+            this._bindModel(this._global.getModel(this._currentLevel));
+        } else {
+            this._globalNotifyHandler = model.connect(`notify::${modelProp}`,
+                this._setCode.bind(this));
+            this._setCode();
+        }
```

Comment:
I think that it would result in a cleaner patch to always connect to `notify::${modelProp}` (or even `notify` I guess, not sure why the pre-existing code did not do that) and add the distinguishing logic to `onGlobalNotify()` instead.

Reply:
`modelProp` is on the per-level model in the pre-existing `perLevel` case, not on the global model. In the`perLevel` case, `_onGlobalNotify()` could actually connect to `notify::currentLevel` I guess, but I didn't want to change that right now.

Reply:
I have not tested this, so we should not take my patch right now, but I pushed this commit to a WIP branch to better illustrate what I meant: https://github.com/endlessm/hack-toolbox-app/commit/1fbca363a15ddd66909e2fbbae49dce76a240258